File: app/src/main/java/space/pal/sig/old/model/database/FactDao.java
package space.pal.sig.old.model.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import io.reactivex.Completable;
import space.pal.sig.old.model.Fact;

/**
 * SpaceNow
 * Created by Catalin on 7/29/2020
 **/
@Dao
public interface FactDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Completable insert(Fact fact);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Completable insert(Fact... facts);

    @Query("delete from facts")
    Completable delete();

    @Query("select * from facts")
    LiveData<List<Fact>> findAll();

}

File: app/src/main/java/space/pal/sig/database/dao/FactDao.kt
package space.pal.sig.database.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import space.pal.sig.model.entity.Fact

/**
 * SpaceNow
 * Created by Catalin on 12/24/2020
 **/
@Dao
interface FactDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun save(fact: Fact)

    @Query("SELECT * FROM facts")
    fun findAll(): LiveData<MutableList<Fact>>

}